### DISCUSSION TITLE
Having issues when styling embla-thumbs__slide class with flex: '0 0 25%'

### DISCUSSION THREAD

**Discussion by arataqueh** [davidjerleke/embla-carousel#539] [Help]
Hello,

I was using the predefined example of the Thumbnail Carousel: https://codesandbox.io/s/c67s5m and whenever I change the slide class from flex: '0 0 100%' to something like below, the carousel gets rendered adding extra spaces on both edges. I am basically following exactly as the codesandbox with the only difference being that the slides are array of ReactNodes instead of numbers. The number of slides are 4 without any gap between them so it should fit perfectly within the container. I am also using the legacy version of next image as ReactNodes so I believe it should not be an issue. What I am missing here?

`'& .embla-thumbs__slide': {
        flex: '0 0 25%',
        minWidth: '0',
      },`

How it gets rendered:
<img width="395" alt="image" src="https://github.com/davidjerleke/embla-carousel/assets/100771928/255c6721-e582-4302-b8bf-e42c8c5aacaa">

Below is my full css(using typescript) for reference:

 '& .embla-thumbs': {
      position: 'relative',
      marginTop: '10px',
    },
    '& .embla-thumbs__viewport': {
      overflow: 'hidden',
    },
    '& .embla-thumbs__container': {
      display: 'flex',
      flexDirection: 'row',
      transform: 'initial',
    },
    '& .embla-thumbs__slide': {
      flex: '0 0 100%',
      minWidth: '100%',
      position: 'relative',
    },
    '& .embla-thumbs__slide__button': {
      '-webkit-appearance': 'none',
      backgroundColor: 'transparent',
      touchAction: 'manipulation',
      display: 'block',
      textDecoration: 'none',
      cursor: 'pointer',
      border: 0,
      padding: 0,
      margin: 0,
      aspectRatio: '16/9',
      width: '100%',
    },
    '& .embla-thumbs__slide--selected .embla-thumbs__slide__button': {
      opacity: 1,
    },
    '& .embla-thumbs__slide__img': {
      display: 'block',
      height: '100%',
      objectFit: 'cover',
    },
    '& .embla__button': {
      position: 'absolute',
      top: '50%',
      transform: 'translateY(-50%)',
    },
    '& .embla__button--next': {
      right: 0,
    },

**Comment by davidjerleke** [comment]
Hi @arataqueh,

Please create a CodeSandbox with **your setup** that **demonstrates the problem** if you want proper guidance. I can’t guess what’s happening based on fragments of code.

Thank you for understanding.

Best,
David

**Reply by arataqueh** [reply]
@davidjerleke I found the solution. Basically I had to force the containScroll: 'trimSnaps' option which is the default value according to the documentation. For those who might face same question, if you use flex percentage for slide sizing, it would create empty slides and this containScroll: 'trimSnaps' would eliminate it.  @davidjerleke if you could update the documentation of the examples explaining  this scenario, I believe it would help people identify this faster, Great library though!

**Reply by davidjerleke** [reply]
@arataqueh I appreciate the effort to help but there's still crucial information missing here.

- What version of Embla are you using? 
- Why didn't you create a CodeSandbox with the problem? That would make it possible for me to see why this was happening.

I don’t want to add information to the docs that might be inaccurate and caused by a faulty setup.

Best,
David
